checkov/terraform/graph_builder/utils.py

```diff
@@ -254,6 +254,15 @@ def attribute_has_nested_attributes(attribute_key: str, attributes: Dict[str, An
     return isinstance(attributes[attribute_key], list) or isinstance(attributes[attribute_key], dict)
 
 
+def attribute_has_dup_with_dynamic_attributes(attribute_key: str, attributes: dict[str, Any] | list[str]) -> bool:
+    attar_key_list = attribute_key.split('.') or [' ', ' ']
```

Comment:
```suggestion
    attr_key_list = attribute_key.split('.')
```
please rename the variable 😉 
the ` or  [...]` will never be triggered, if you split a string with no `.` inside it will return a list with the original value. you can then check, if `len(attr_key_list) > 1` otherwise return `False`